xdsl/parser/base_parser.py

```diff
@@ -45,7 +45,7 @@ def parse_optional_integer(
         """
         Parse an (possible negative) integer. The integer can either be
         decimal or hexadecimal.
-        Optionally allow parsing of 'true' or 'false' into 1 and 0.
+        Optionally allow parsing of 'true' or 'false' into -1 and 0.
```

Comment:
This does not agree with the implementation below

Reply:
It does though

Reply:
Ah actually not here

Reply:
You're right, all of the functions with the changed documentation here return the 1/0 values, it's only when creating the attribute that the value is different. This should  be reflected in documentation for the attribute, not the parser.